### PR TITLE
🚡 upgrade node version to 16

### PR DESCRIPTION
**Pull request by siddhesh1770**

### CODE REVIEW

**Comment by Moshyfawn** [comment]
Reason?
Related: #8971

**Comment by bluebill1049** [comment]
thanks for the PR, honestly we should upgrade to 16 not downgrade.

**Comment by Moshyfawn** [comment]
Wow, how the turn tables lol

**Comment by Moshyfawn** [comment]
Perhaps the LTS version? [v16.13.0 (LTS)](https://nodejs.org/ru/blog/release/v16.13.0/)

**Comment by bluebill1049** [comment]
circle ci shitted itself :(

**Comment by bluebill1049** [comment]
let me try again

**Comment by bluebill1049** [comment]
Screen Shot 2022-09-10 at 9 22 29 am

:(

**Comment by bluebill1049** [comment]
fixing CSB config as well

**Comment by Moshyfawn** [comment]
Oh, I had to fix this node version CI issue recently in Cloudflare. Had to set the NODE_VERSION var to the proper Node version. Haven't experienced the same issue with CircleCI, so I wouldn't know where to put it
image

**Comment by bluebill1049** [comment]
all good 👍 back on track

**Comment by Moshyfawn** [comment]
I wonder if it needs a major release, tho..

**Comment by bluebill1049** [comment]
think twice perhaps we don't need the node engine attribute?

**Comment by Moshyfawn** [comment]
Hmm..

**Comment by Moshyfawn** [comment]
We're using the `spread operator` and `es2018` is defined in the `tsconfig`, so, I believe, we need at least `v10.13.0`. React's lowest Node version is `12`..

**Comment by bluebill1049** [comment]
> We're using the `spread operator` and `es2018` is defined in the `tsconfig`, so, I believe, we need at least `v10.13.0`. React's lowest Node version is `12`..

You are right, i am going to close this PR for now then. thanks for the help.
